polling/pcc_policy_config.go

```diff
@@ -0,0 +1,225 @@
+// SPDX-FileCopyrightText: 2025 Canonical Ltd
+
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package polling
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/omec-project/openapi/models"
+	"github.com/omec-project/openapi/nfConfigApi"
+	"github.com/omec-project/pcf/logger"
+	"github.com/omec-project/util/idgenerator"
+)
+
+var (
+	pccPolicies map[models.Snssai]*PccPolicy
+	configLock  sync.RWMutex
+)
+
+type PccPolicy struct {
+	PccRules      map[string]*models.PccRule
+	QosDecs       map[string]*models.QosData
+	TraffContDecs map[string]*models.TrafficControlData
+}
+
+var GetSlicePccPolicy = func(snssai models.Snssai) *PccPolicy {
+	configLock.RLock()
+	defer configLock.RUnlock()
+	slicePccPolicyData, ok := pccPolicies[snssai]
+	if ok {
+		return slicePccPolicyData
+	}
+	return nil
+}
```

Comment:
I don't think this is getting mocked in tests, because they are in the same package and can change `pccPolicies` instead. I would change this to a regular `func`.